cmd/builder/README.md

```diff
@@ -145,3 +145,44 @@ then commit the code in a git repo. A CI can sync the code and execute
 ocb --skip-generate --skip-get-modules --config=config.yaml
 ```
 to only execute the compilation step.
+
+### Avoiding the use of a new go.mod file
+
+There is an additional option that controls one aspect of the build
+process, which specifically allows skipping the use of a new `go.mod`
+file.  When the `--skip-new-go-module` command-line flag is supplied,
+the build process issues a `go get` command for each component,
+relying on the Go toolchain to update the enclosing Go module
+appropriately.
+
+This command will avoid downgrading a dependency in the enclosing
+module, according to
+[`semver.Compare()`](https://pkg.go.dev/golang.org/x/mod/semver#Compare),
+and will instead issue a log indicating that the component was not
+upgraded.
+
+This mode cannot be used in conjunction with several features that
+control the generated `go.mod` file, including `replaces`, `excludes`,
+and the component `path` override.  For each of these features, users
+are expected to modify the enclosing `go.mod` directly.
+
+### Strict versioning checks
+
+There is an option that controls version strictness applied by the
+builder.  When the `--strict-versioning` command-line flag is used,
+the following additional checks apply to the finished `go.mod` file
+after getting all the components and tidying the result. 
+
+1. The `dist::otelcol_version` field in the build configuration must
+   match the core library version calculated by the Go toolchain,
+   considering all components.  A mismatch could happen, for example,
+   when one of the components depends on a newer release of the core
+   collector library.
+2. For each component in the build configuration, the version included
+   in the `gomod` module specifier must match the one calculated by
+   the Go toolchain, considering all components.  A mismatch could
+   happen, for example, when the enclosing Go module uses a newer
+   release of the core collector library.
```

Comment:
I think this should be the default, and would have prevented issues such as #8692. Let's do this on a separate PR, I am happy to review that one :)